packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -27,10 +28,14 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
     super();
     this.metadata = metadata;
     this.preference = preference;
-    this.communicator = new Communicator(keysUrl, metadata);
+
+    const url = typeof keysUrl === 'string' ? keysUrl : CB_KEYS_URL;
```

Comment:
Removing keysUrl as an option resulted in a type error here. Now we're type checking for a string before passing to the communicator. Might be worth allowing unknown and validating inside the communicator

Reply:
i see, actually maybe good to have a separate PR to remove keysUrl? i can take it on, P2/P3 maybe?

Reply:
we can revert all keysUrl change in this PR? i'll tackle it separately